runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/StreamingDataflowWorker.java

```diff
@@ -245,17 +247,34 @@ private StreamingDataflowWorker(
     Consumer<PrintWriter> getDataStatusProvider;
     Supplier<Long> currentActiveCommitBytesProvider;
     ChannelCache channelCache = null;
+    WindmillStreamPool<GetDataStream> getDataStreamPool = null;
     if (options.isEnableStreamingEngine() && options.getIsWindmillServiceDirectPathEnabled()) {
       // Direct path pipelines.
       WeightedSemaphore<Commit> maxCommitByteSemaphore = Commits.maxCommitByteSemaphore();
       channelCache = createChannelCache(options, configFetcher);
+      getDataStreamPool =
```

Comment:
all of these deps that are just for the fallback case I'm wondering if we can create lazily so that we don't startup threads etc unless we need to.

One maybe easy way to do that is replace all the ones that are only used in the fallback path with a Supplier and Suppliers.memoize

example
Supplier<WindmillStreamPool> ... = Suppliers.memoize(...);
Supplier<GetDataClient> ... = Supppliers.memoize(()->new StreamPoolGetDataClient(getDataMetricTracker, getDataStreamPool.get());